napari/_qt/containers/qt_axis_model.py

```diff
@@ -0,0 +1,164 @@
+
+        The AxisList is filled with a number of AxisModels based
+        on the number of dimensions in the Dims object.
+
+        Parameters
+        ----------
+        dims : napari.components.dims.Dims
+            Dims object to be used for creation.
+
+        Returns
+        -------
+        AxisList
+            A selectable evented list of the viewer axes.
+        """
+        return cls([AxisModel(dims, dims.order[i]) for i in range(dims.ndim)])
+
+
+class QtAxisListModel(QtListModel[AxisModel]):
+    def data(self, index: QModelIndex, role: Qt.ItemDataRole):
+        if not index.isValid():
+            return None
+        axis = self.getItem(index)
+        if role == Qt.ItemDataRole.DisplayRole:
+            return str(axis)
+        if role == Qt.ItemDataRole.TextAlignmentRole:
+            return Qt.AlignCenter
+        if role == Qt.ItemDataRole.CheckStateRole:
+            return (
+                Qt.CheckState.Checked
+                if axis.rollable
+                else Qt.CheckState.Unchecked
+            )
+        return super().data(index, role)
+
+    def setData(
+        self,
+        index: QModelIndex,
+        value: Any,
+        role: int = Qt.ItemDataRole.EditRole,
+    ) -> bool:
+        axis = self.getItem(index)
+        if role == Qt.ItemDataRole.CheckStateRole:
+            axis.rollable = bool(
+                value == Qt.CheckState.Checked or value is True
+            )
+        elif role == Qt.ItemDataRole.EditRole:
+            axis_labels = list(axis.dims.axis_labels)
+            axis_labels[axis.axis] = value
+            axis.dims.axis_labels = tuple(axis_labels)
+        else:
+            return super().setData(index, value, role=role)
+        self.dataChanged.emit(index, index, [role])
+        return True
+
+    def flags(self, index: QModelIndex) -> Qt.ItemFlags:
+        """Returns the item flags for the given `index`.
+
+        This describes the properties of a given item in the model.  We set
+        them to be editable, checkable, draggable, droppable, etc...
+        If index is not a list, we additionally set `Qt.ItemNeverHasChildren`
+        (for optimization). Editable models must return a value containing
+        `Qt.ItemIsEditable`.
+
+        See Qt.ItemFlags https://doc.qt.io/qt-5/qt.html#ItemFlag-enum
+
+        Parameters
+        ----------
+        index : qtpy.QtCore.QModelIndex
+            Index to return flags for.
+
+        Returns
+        -------
+        qtpy.QtCore.Qt.ItemFlags
+            ItemFlags specific to the given index.
+        """
+        flags = (
+            Qt.ItemFlag.ItemIsSelectable
+            | Qt.ItemFlag.ItemIsEditable
+            | Qt.ItemFlag.ItemIsUserCheckable
+            | Qt.ItemFlag.ItemIsEnabled
+            | Qt.ItemFlag.ItemNeverHasChildren
+        )
+
+        if not index.isValid():
+            # we allow drops outside the items
+            return Qt.ItemFlag.ItemIsDropEnabled
```

Comment:
Why we plan to allow drops from outside? Could you describe uscase? We have only one Axis list?

Reply:
Ah, I think was copied over from `qt_layer_model` (or rather `_base_item_model`). So agreed this can be removed.

Reply:
Nope, I'm wrong. Outside here means not inside each item. It does not mean from outside this list. I.e. either before any items (at the top), in between two items, or after all items (at the bottom).

The QListView that owns this model only defines internal moves for drag and drop: https://github.com/napari/napari/blob/57754c4714572a6e0602029a91eb7be0ad07644a/napari/_qt/containers/qt_list_view.py#L44